[Doc Improvement][Microsoft Entra ID for Teams apps][2896821] (#10037)
MicrosoftDocs/msteams-docs: msteams-platform/toolkit/create-resource-bot-microsoft-entra-id.md

---
title: Create bot resource in Microsoft Entra ID
author: surbhigupta
description: Create an Azure bot resource in Microsoft Entra ID to add your bot app to Teams
ms.topic: how-to
ms.author: surbhigupta
ms.localizationpriority: medium
---
# Create a bot resource in Microsoft Entra ID

> Your Teams bot app can access resources available in Microsoft Entra ID. You need to create a bot resource with Microsoft Entra ID. This process registers your bot app with Azure AI Bot Service and enable the bot app to connect to Teams channels.

Reviewer: ```suggestion
For your Teams bot app to access resources available in Microsoft Entra ID, you'll need to create a bot resource with Microsoft Entra ID. The following process registers your bot app with Azure AI Bot Service and enables the app to connect to Teams channels.
```

> 

Reviewer: Can we add a list of prerequisites the user may need to complete tutorial.

[Example](https://review.learn.microsoft.com/en-us/entra/identity-platform/quickstart-web-app-dotnet-core-sign-in?branch=main).

Author: Done.

> In this section, you will:

Reviewer: ```suggestion
In this section, you'll:
```

- [Create and deploy bot resource](#create-a-bot-resource-in-microsoft-entra-id)
- [Configure the client secret](#create-client-secret)
- [Enable the bot resource for Teams](#enable-bot-for-teams)

## Create and deploy bot resource in Microsoft Entra ID

You can create a bot resource in Microsoft Entra ID for a single or multitenant app.

### To create and deploy bot resource

1. Open the [Azure portal](https://ms.portal.azure.com/) on your web browser.

1. Select the **Create a resource** icon.

    :::image type="content" source="../assets/images/meetings-side-panel/create-resource-azure-portal.png" alt-text="Screenshot shows the create a resource option in Azure portal.":::

    The **Create a resource** page appears.

1. Type **Azure bot** in the search box, and select the Azure bot from the options that appear.

    :::image type="content" source="../assets/images/authentication/select-azure-bot-marketplace.png" alt-text="Screenshot shows the Azure bot in the Azure Marketplace.":::

>     The **Azure Bot** page appears, and the plan is preselected as Azure Bot.

Reviewer: ```suggestion
    When the **Azure Bot** window appears, ensure the selected plan is **Azure Bot**.
```

1. Select **Create**.

    :::image type="content" source="../assets/images/authentication/azure-bot-create.png" alt-text="Screenshot shows the Azure bot plan selected in the Azure bot creation screen.":::

    The **Create an Azure Bot** page appears.

1. Under **Project details**, enter the following details:

>     1. Enter a unique identifier as the bot handle. It isn't the display name, and you can choose a different display name later.

Reviewer: ```suggestion
    1. Enter a unique identifier as the bot handle, such as _azure-teams-bot_. This isn't the display name, and changed at a later point.
```
This doesn't have to be the name, but it's good to give an example.

        1. Enter a name for the resource and select **OK**.

1. Under **Pricing**, the standard price tier is selected by default. You can change your pricing tier by clicking on **Change plan**.

1. Under **Microsoft App ID**, you need to configure the type of app and the creation type.

    1. Depending on how you want to use the bot resource, choose from **User-assigned Managed Identity**, **Multi Tenant**, and **Single Tenant** as the type of app.

    1. For the app creation type, you can either create a new Microsoft App ID or use an existing app registration. If you select **Create new Microsoft App ID**, Azure creates a new and unique app ID for your bot resource.

    1. If you already have a registered app that you want to link the bot resource to, select **Use existing app registration** as the creation type.

        :::image type="content" source="../assets/images/authentication/select-existing-app-registration.png" alt-text="Screenshot shows the option to enter an existing app ID in the create Azure bot screen. ":::

        The fields for entering the app ID details appear.

    1. Enter the app ID of the Microsoft Entra app you have already registered.

> 

Reviewer: ```suggestion
1. Select **Next** .
```

Author: Done.

> 1. Select **Next** and the **Tags** tab opens. This step is optional and tags can be configured from the bot resource's overview page anytime.

Reviewer: ```suggestion
1. When the **Tags** tab opens. This step is optional and tags can be configured from the bot resource's overview page anytime.
```

    1. Enter the name and value tags for categorizing resources you provisioned.

    :::image type="content" source="../assets/images/authentication/name-value-tag-azure-bot.png" alt-text="Screenshot shows the tags tab in the create an Azure bot screen.":::

1. Select **Review + create**. Microsoft Entra ID validates the entered details.

1. Review your bot resource and select **Create** to create the bot.

    :::image type="content" source="../assets/images/authentication/resource-bot-creation-final-preview.png" alt-text="Screenshot shows the final preview screen before creating a bot in Azure portal.":::

     A message pops up on the browser stating that the deployment is being initialized.

    :::image type="content" source="../assets/images/authentication/initialize-deployment-message.png" alt-text="Screenshot shows a message stating the deployment of the resource is initializing.":::

>     The **Overview** page opens as the deployment is in progress. Once the deployment is complete, you get a **Deployment succeeded** message. This means that the bot resource is successfully created and the selected resources are provisioned.

Reviewer: ```suggestion
    The **Overview** page opens as the deployment is in progress. Once the deployment is complete, a**Deployment succeeded** message is displayed. This message confirms that the bot resource has been created successfully and the selected resources provisioned.
```

1. Under **Next steps** dropdown, select **Go to resource**.

    :::image type="content" source="../assets/images/authentication/go-to-bot-resource.png" alt-text="Screenshot shows the Azure bot resource deployment page.":::

>      You can see the details of the bot resource you deployed.

Reviewer: ```suggestion
     The bot resource details are displayed on this page.
```

    :::image type="content" source="../assets/images/authentication/bot-resource-overview-page.png" alt-text="Screenshot shows the overview page of the resource bot":::

> After you create your bot resource, you need to add a client secret and enable the bot to work in Teams.

Reviewer: ```suggestion
After you create your bot resource, you need to add a client secret and enable the bot to work in Teams.
```
Can you move this into the blow H3 ### Create client secret

     The **Certificates & secrets** page appears.

2. Select **+ New client secret**.

    :::image type="content" source="../assets/images/authentication/client-secret.png" alt-text="Screenshot shows the add a client secret option highlighted in red.":::

   The **Add a client secret** page appears.

    :::image type="content" source="../assets/images/authentication/add-a-client-secret.png" alt-text="Screenshot shows the add a client secret page.":::

3. Enter the description, select the duration of validity for the secret, and select **Add**.

    A message pops up on the browser stating that the credentials of the bot resource were updated. The client secret displays on the page.

    :::image type="content" source="../assets/images/authentication/client-secret-updated.png" alt-text="Screenshot shows the client secret of the bot resource.":::

> [!NOTE]
> Ensure that you copy the value of client secret right after you create it. The value is visible only at the time when the client secret is created, and can't be viewed after that.

### Enable bot for Teams

> You must enable the Teams channel to let the bot interact with Microsoft Teams.

Reviewer: We'll need to clarity here as to why we need to enable the Teams channel, not sure what this sentence achieves currently.
